packages/serverpod_serialization/lib/src/duration_ext.dart

```diff
@@ -0,0 +1,98 @@
+/// Extension for decoding [Duration] in the protocol.
+extension DurationDec on String {
```

Comment:
@yahu1031 Does intl library have parser from String to Duration?
If yes, could you share the method name or link to the docs to that method?
Thank you

Reply:
You could also just use the `inMicroseconds` property, store the value as an `int`, recreate the `Duration` from the microseconds. That seems much better than involving the `intl` package.

Reply:
I am afraid that when the user wants to insert ```Duration(days: 9999999999)```, it can be stored incorrectly while inMicroseconds or inMilliSeconds are used

Reply:
2^63 - 1 is the max value of int in Dart, that should be enough, no?

Reply:
That's 213503982 days.

Reply:
Or 584942 years. I doubt anyone use more than that for calculating dates and times.

Reply:
213503982 days * 8,64e+7 miliseconds=very big number😅

Reply:
Also, just checked. The Duration is internally stored as an `int` anyway. So it cannot handle any higher values.

Reply:
In that case, I have no argument )
If you request to change it to int, ill update my mr